pkg/scheduler/plugins/elasticquota/core/group_quota_manager.go

```diff
@@ -0,0 +1,616 @@
+			return nil
+		}
+
+		inputQuotaGroups = append(inputQuotaGroups, quotaInfo)
+	}
+	return gqm.updateQuotaGroupConfigNoLock(inputQuotaGroups)
+}
+
+func (gqm *GroupQuotaManager) updateQuotaGroupConfigNoLock(inputGroups []*QuotaInfo) error {
+	//build allNodes based on the input
+	newAllQuotaTopoNodes := gqm.buildSubParGroupTopoNoLock(inputGroups)
+
+	gqm.resetAllGroupQuotaNoLock(newAllQuotaTopoNodes)
+	return nil
+}
+
+//BuildSubParGroupTopoNoLock reBuild a nodeTree from root, no need to lock gqm.lock
+func (gqm *GroupQuotaManager) buildSubParGroupTopoNoLock(updateGroups []*QuotaInfo) map[string]*QuotaTopoNode {
+	//rebuild tree
+	allQuotaTopoNodes := make(map[string]*QuotaTopoNode)
+	rootNode := NewQuotaTopoNode(NewQuotaInfo(true, true, extension.RootQuotaName, extension.RootQuotaName))
+	allQuotaTopoNodes[extension.RootQuotaName] = rootNode
+
+	//get already created node
+	for quotaName, quotaInfo := range gqm.quotaInfoMap {
+		allQuotaTopoNodes[quotaName] = NewQuotaTopoNode(quotaInfo)
+	}
+	//get newCreated nodes
+	for _, quotaInfo := range updateGroups {
+		if _, exist := allQuotaTopoNodes[quotaInfo.Name]; !exist {
+			topoNode := NewQuotaTopoNode(&QuotaInfo{
+				Name: quotaInfo.Name,
+			})
+			allQuotaTopoNodes[quotaInfo.Name] = topoNode
+		}
+	}
+	for _, quotaInfo := range updateGroups {
+		updateQuotaInfo := allQuotaTopoNodes[quotaInfo.Name].quotaInfo
+		// update the quotaNode's quotaInfo
+		updateQuotaInfo.UpdateQuotaInfoFromRemote(quotaInfo)
+	}
+
+	// build tree according to the parGroupName
+	for _, topoNode := range allQuotaTopoNodes {
+		if topoNode.groupName == extension.RootQuotaName {
+			continue
+		}
+		parQuotaTopoNode := allQuotaTopoNodes[topoNode.quotaInfo.ParentName]
+		topoNode.parQuotaTopoNode = parQuotaTopoNode
+		parQuotaTopoNode.AddChildGroupQuotaInfo(topoNode)
+	}
+
+	return allQuotaTopoNodes
+}
+
+// ResetAllGroupQuotaNoLock no need to lock gqm.lock
+func (gqm *GroupQuotaManager) resetAllGroupQuotaNoLock(newAllQuotaTopoNodes map[string]*QuotaTopoNode) {
+	newMaxQuotaMap, newMinQuotaMap, newSharedWeightMap, childRequestMap, childUsedMap := make(quotaResMapType), make(quotaResMapType),
+		make(quotaResMapType), make(quotaResMapType), make(quotaResMapType)
+	for quotaName, topoNode := range newAllQuotaTopoNodes {
+		if quotaName == extension.RootQuotaName {
+			continue
+		}
+		topoNode.quotaInfo.lock.Lock()
+		newMaxQuotaMap[quotaName] = topoNode.quotaInfo.CalculateInfo.Max.DeepCopy()
+		newMinQuotaMap[quotaName] = topoNode.quotaInfo.CalculateInfo.OriginalMin.DeepCopy()
+		newSharedWeightMap[quotaName] = topoNode.quotaInfo.CalculateInfo.SharedWeight.DeepCopy()
+
+		topoNode.quotaInfo.setMaxQuotaNoLock(v1.ResourceList{})
+		topoNode.quotaInfo.setOriginalMinQuotaNoLock(v1.ResourceList{})
+		topoNode.quotaInfo.setSharedWeightNoLock(v1.ResourceList{})
+
+		if !topoNode.quotaInfo.IsParent {
+			childRequestMap[quotaName] = topoNode.quotaInfo.CalculateInfo.Request.DeepCopy()
```

Comment:
if quota is from update logic before, due to  "quotaInfo := NewQuotaInfoFromQuota(quota)", you can't get the correct old request and used,  I am worried how do you pass you ut?